### DISCUSSION TITLE
How to customize spacing at different media queries?

### DISCUSSION THREAD

**Discussion by dynamite-bud** [tailwindlabs/tailwindcss#9144] [Help] [answered]
The app at our company currently uses SCSS but I am migrating it to tailwind. How to make spacing different on different media queries like
If I customize spacing at p-1 for 0.25 rem but i want to change spacing for p-1 to 0.20 rem at media query how do i do that ?

Our SCSS file :

$val: 0.25rem;

@for $i from 0 through 10 {
	.p-#{$i} {
		padding: $i * $val !important;
	}
	.p-#{$i}-l {
		padding-left: $i * $val !important;
	}

	.p-#{$i}-r {
		padding-right: $i * $val !important;
	}

	.p-#{$i}-t {
		padding-top: $i * $val !important;
	}

	.p-#{$i}-b {
		padding-bottom: $i * $val !important;
	}

	.p-#{$i}-lr {
		padding-left: $i * $val !important;
		padding-right: $i * $val !important;
	}

	.p-#{$i}-tb {
		padding-top: $i * $val !important;
		padding-bottom: $i * $val !important;
	}

	.m-#{$i} {
		margin: $i * $val !important;
	}
	.m-#{$i}-l {
		margin-left: $i * $val !important;
	}

	.m-#{$i}-r {
		margin-right: $i * $val !important;
	}

	.m-#{$i}-t {
		margin-top: $i * $val !important;
	}

	.m-#{$i}-b {
		margin-bottom: $i * $val !important;
	}

	.m-#{$i}-lr {
		margin-left: $i * $val !important;
		margin-right: $i * $val !important;
	}

	.m-#{$i}-tb {
		margin-top: $i * $val !important;
		margin-bottom: $i * $val !important;
	}
}

@media screen and (max-width: 1440px), screen and (max-height: 940px) {
	$val: 0.2rem;

	/* Padding css */
	@for $i from 1 through 10 {
		.p-#{$i} {
			padding: $i * $val !important;
		}
		.p-#{$i}-l {
			padding-left: $i * $val !important;
		}

		.p-#{$i}-r {
			padding-right: $i * $val !important;
		}

		.p-#{$i}-t {
			padding-top: $i * $val !important;
		}

		.p-#{$i}-b {
			padding-bottom: $i * $val !important;
		}

		.p-#{$i}-lr {
			padding-left: $i * $val !important;
			padding-right: $i * $val !important;
		}

		.p-#{$i}-tb {
			padding-top: $i * $val !important;
			padding-bottom: $i * $val !important;
		}

		.m-#{$i} {
			margin: $i * $val !important;
		}
		.m-#{$i}-l {
			margin-left: $i * $val !important;
		}

		.m-#{$i}-r {
			margin-right: $i * $val !important;
		}

		.m-#{$i}-t {
			margin-top: $i * $val !important;
		}

		.m-#{$i}-b {
			margin-bottom: $i * $val !important;
		}

		.m-#{$i}-lr {
			margin-left: $i * $val !important;
			margin-right: $i * $val !important;
		}

		.m-#{$i}-tb {
			margin-top: $i * $val !important;
			margin-bottom: $i * $val !important;
		}
	}
}

**Comment by wongjn** [comment, marked as answer]
The "pure" Tailwind way would be to use another class, so `p-[.2rem] big:p-1` for example. If you would use this `0.2rem` often, then you can look at [extending the spacing scale](https://tailwindcss.com/docs/customizing-spacing#extending-the-default-spacing-scale). The `big:` would be a media query variant defined in your `tailwind.config.js` like so:

```js
module.exports = {
  theme: {
    extend: {
      screens: {
        big: { raw: 'screen and (min-width: 1441px), screen and (min-height: 941px)' },
      },
    },
  },
};
```

Notice also that I have adapted to be `min-` since Tailwind uses the minimum breakpoint system and works its way up. If you wanted to strictly follow your existing code of `max-` breakpoints, then the class could be like `p-1 smaller:p-[.2rem]` with the following config:

```js
module.exports = {
  theme: {
    extend: {
      screens: {
        smaller: { raw: 'screen and (max-width: 1440px), screen and (max-height: 940px)' },
      },
    },
  },
};
```

---

If you'd like to reuse your current system where it is a single class that adapts automatically, then perhaps look at using CSS variables:

```js
module.exports = {
  theme: {
    extends: {
      spacing: {
        // To keep things DRY, you could look at
        // generating these in a loop.
        '1': 'var(--val)',
        '2': 'calc(var(--val) * 2)',
        // …
      }.
    },
  },
};
```

Then in your CSS:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --val: 0.25rem;
  }
  @meda screen and (max-width: 1440px), screen and (max-height: 940px) {
    :root {
      --val: 0.2rem;
    }
  }
}
```

**Reply by dynamite-bud** [reply]
Thank you 🙏🏼
